src/main.cpp

```diff
@@ -4693,47 +4693,50 @@ bool AcceptBlock(CBlock& block, CValidationState& state, CBlockIndex** ppindex,
             // Start at the block we're adding on to
             CBlockIndex *prev = pindexPrev;
 
-            int readBlock = 0;
             vector<CBigNum> vBlockSerials;
-            CBlock bl;
-            // Go backwards on the forked chain up to the split
-            do {
-                // Check if the forked chain is longer than the max reorg limit
-                if(readBlock == Params().MaxReorganizationDepth()){
-                    // TODO: Remove this chain from disk.
-                    return error("%s: forked chain longer than maximum reorg limit", __func__);
-                }
-
-                if(!ReadBlockFromDisk(bl, prev))
-                    // Previous block not on disk
-                    return error("%s: previous block %s not on disk", __func__, prev->GetBlockHash().GetHex());
-                // Increase amount of read blocks
-                readBlock++;
-                // Loop through every input from said block
-                for (const CTransaction& t : bl.vtx) {
-                    for (const CTxIn& in: t.vin) {
-                        // Loop through every input of the staking tx
-                        for (const CTxIn& stakeIn : pivInputs) {
-                            // if it's already spent
-
-                            // First regular staking check
-                            if(hasPIVInputs) {
-                                if (stakeIn.prevout == in.prevout) {
-                                    return state.DoS(100, error("%s: input already spent on a previous block", __func__));
-                                }
+            if (!chainActive.Contains(prev)) {
```

Comment:
Doesn't this _always_ fail?
At this point we've found `pindexPrev` somewhere in the blockmap and it's != the current tip, so `[pindex->nHeight] == pindex;` would be always false.The implementation of `Contains()` is a bit confusing for me.

Reply:
`Contains()` checks if the block is on the main chain or not.

pindexPrev here is the previous block of the forked block that just arrived, it could or not be on chainActive vector. The only scenario in which this first check is true and don't start the loop that finds the split block is when the fork is just one block ahead.

Example, the main chain is in block 300:
1) block_A 301 arrives and it's connected.
2) block_B 301 (fork) arrives, this block has block 300 as pindexPrev which is on the main chain and will return true for the `chainActive.Contains(prev)`.